app/views/handlers/list_action.py

```diff
@@ -52,21 +52,31 @@ def get_next_location_url(self):
             ):
                 return self.get_section_summary_url()
 
+        if self.router.is_block_complete(
+            block_id=self.parent_location.block_id,
+            section_id=self.parent_location.section_id,
+            list_item_id=self.parent_location.list_item_id,
+        ):
+            return self.router.get_next_location_url(
+                self.parent_location,
+                self._routing_path,
+                self._return_to,
+                self._return_to_answer_id,
+                self._return_to_block_id,
+            )
+
         return self.parent_location.url()
 
     def handle_post(self):
         self.questionnaire_store_updater.update_same_name_items(
             self.parent_block["for_list"],
             self.parent_block.get("same_name_answer_ids"),
         )
-        # Clear the answer from the confirmation question on the list collector question
-        answer_ids_to_remove = self._schema.get_answer_ids_for_block(
-            self.parent_location.block_id
-        )
-        self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
-        self.evaluate_and_update_section_status_on_list_change(
-            self.parent_block["for_list"]
+
+        self._routing_path = self.router.routing_path(
+            self.current_location.section_id, self.current_location.list_item_id
         )
+
         self.questionnaire_store_updater.save()
```

Comment:
We should only need to do this if any store has been updated.

```suggestion
        if self.questionnaire_store_updater.is_dirty():
            self._routing_path = self.router.routing_path(
                self.current_location.section_id, self.current_location.list_item_id
            )

            self.questionnaire_store_updater.save()
```